ambry-router/src/test/java/com/github/ambry/router/GetBlobOperationTest.java

```diff
@@ -897,6 +901,39 @@ public void testOrigDcUnavailability() throws Exception {
     getErrorCodeChecker.testAndAssert(RouterErrorCode.AmbryUnavailable);
   }
 
+  /**
+   * Test that if local DC has less than {@code RouterConfig#routerGetOperationMinLocalReplicaCountToPrioritizeLocal}
+   * replicas available, then we will prioritize remote DC.
+   * @throws Exception
+   */
+  @Test
+  public void testGetGoesToRemoteIfLocalDcHasFewAvailableReplicas() throws Exception {
+    doPut();
+    Properties props = getDefaultNonBlockingRouterProperties(true);
+    props.setProperty(RouterConfig.ROUTER_GET_ELIGIBLE_REPLICAS_BY_STATE_ENABLED, "true");
+    props.setProperty(RouterConfig.ROUTER_GET_BLOB_RETRY_LIMIT_COUNT, "10");
+    props.setProperty(RouterConfig.ROUTER_GET_BLOB_RETRY_LIMIT_IN_SEC, "10");
+    props.setProperty(RouterConfig.ROUTER_GET_OPERATION_MIN_LOCAL_REPLICA_COUNT_TO_PRIORITIZE_LOCAL, "1");
+    RouterConfig oldRouterConfig = routerConfig;
+    routerConfig = new RouterConfig(new VerifiableProperties(props));
+
+    try {
+      // For originating DC, let's bring down two replicas
+      MockPartitionId partitionId = (MockPartitionId) blobId.getPartition();
+      List<ReplicaId> localReplicas = partitionId.replicaIds.stream()
+          .filter(replicaId -> replicaId.getDataNodeId().getDatacenterName().equals(localDcName))
+          .collect(Collectors.toList());
+      partitionId.setReplicaState(localReplicas.get(0), ReplicaState.LEADER);
+      partitionId.setReplicaState(localReplicas.get(1), ReplicaState.OFFLINE);
+      partitionId.setReplicaState(localReplicas.get(2), ReplicaState.OFFLINE);
+
+      getAndAssertSuccess();
+      assertTrue(routerMetrics.shuffledWithRemoteReplicasForGetDueToFewLocalReplicas.getCount() > 0);
```

Comment:
A better assertion would be get the value before calling `getAndAssertSuccess` and the value after `getAndAssertSuccess` and make sure the value after = value before + 1.

Reply:
Its actually a little more complicated. So the number of times the counter `shuffledWithRemoteReplicasForGetDueToFewLocalReplicas` is incremented depends on number of chunks in a blob and if that chunk was placed in the partition that has doesn't have enough replicas in the local colo.

In order to make the test more deterministic, and also to not introduce unnecessary complication, I have changed the test to use simple blobs only.